R/batchmark.R

```diff
@@ -0,0 +1,182 @@
+    data.table(i = seq_len(rdesc$iters))
+  }, id = task.ids, task = tasks, rdesc = resamplings, seed = reg$seed + seq_along(tasks))
+
+  # generate algos
+  ades = Map(function(id, learner) {
+    apply.fun = getAlgoFun(learner, measures, models)
+    batchtools::addAlgorithm(id, apply.fun, reg = reg)
+    data.table()
+  }, id = learner.ids, learner = learners)
+
+  # add experiments
+  batchtools::addExperiments(reg = reg, prob.designs = pdes, algo.designs = ades, repls = 1L)
+}
+
+resample.fun = function(job, data, i) {
+  if (is.numeric(data$task))
+    data$task = OpenML::convertOMLDataSetToMlr(getOMLDataSet(data$task))
+  rin = makeResampleInstance(desc = data$rdesc, task = data$task)
+  list(train = rin$train.inds[[i]], test = rin$test.inds[[i]], weights = rin$weights[[i]])
+}
+
+getAlgoFun = function(lrn, measures, models) {
+  force(lrn)
+  force(measures)
+  force(models)
+  function(job, data, instance) {
+
+    extract.this = getExtractor(lrn)
+    if (is.numeric(data$task))
+      data$task = OpenML::convertOMLDataSetToMlr(getOMLDataSet(data$task))
+
+    calculateResampleIterationResult(learner = lrn, task = data$task, train.i = instance$train, test.i = instance$test, 
+      measures = measures, weights = instance$weights, rdesc = data$rdesc, model = models, extract = extract.this)
+  }
+}
+
+#' @title Reduce results of a batch-distributed benchmark.
+#' 
+#' @description
+#' This creates a \code{\link{BenchmarkResult}} from a \code{\link[batchtools]{Registry}}.
+#' To setup the benchmark have a look at \code{\link{batchmark}}.
+#' 
+#' @param ids [\code{data.frame} or \code{integer}]\cr
+#'   A \code{data.frame} (or \code{\link[data.table]{data.table}})
+#'   with a column named \dQuote{job.id}.
+#'   Alternatively, you may also pass a vector of integerish job ids.
+#'   If not set, defaults to all jobs.
+#' @template arg_keep_pred
+#' @template arg_showinfo 
+#' @param reg [\code{\link[batchtools]{Registry}}]\cr
+#'   Registry, created by \code{\link[batchtools]{makeExperimentRegistry}}. If not explicitly passed, 
+#'   uses the last created registry.
+#' @return [\code{\link{BenchmarkResult}}].
+#' @export
+#' @family benchmark
+reduceBatchmarkResults = function(ids = NULL, keep.pred = TRUE, show.info = getMlrOption("show.info"), reg = batchtools::getDefaultRegistry()) {
+  
+  #registry and ids are asserted later
+  requirePackages("batchtools", why = "batchmark", default.method = "load")
+  assertFlag(keep.pred)
+  
+  result = list()
+  
+  problems = batchtools::getProblemIds(reg)
+  algorithms = batchtools::getAlgorithmIds(reg)
+  
+  for (p in problems) {
+    
+    exps = batchtools::findExperiments(prob.name = p, ids = ids)
+    
+    if (nrow(exps) > 0) {
+      problem = batchtools::makeJob(id = exps[1, ])$problem
+      pname = p
+      if (is.numeric(problem$data$task)) {
+        problem$data$task = OpenML::convertOMLDataSetToMlr(getOMLDataSet(problem$data$task))
```

Comment:
Same argument as earlier. Also we cannot keep it in memory since the `reduceBatchmarkResults` may be called in a different/later R session. Again, we get the dataset from the OML cache instead of saving it in the registry.